arbnode/dataposter/data_poster.go

```diff
@@ -378,13 +379,26 @@ func (p *DataPoster) evalMaxFeeCapExpr(backlogOfBatches uint64, elapsed time.Dur
 		"ElapsedTimeBase":       float64(config.ElapsedTimeBase),
 		"ElapsedTimeImportance": config.ElapsedTimeImportance,
 		"TargetPriceGWei":       config.TargetPriceGwei,
-		"GWei":                  params.GWei,
 	}
 	result, err := p.maxFeeCapExpression.Evaluate(parameters)
 	if err != nil {
 		return nil, fmt.Errorf("error evaluating maxFeeCapExpression: %w", err)
 	}
-	return arbmath.FloatToBig(result.(float64)), nil
+	resultFloat, ok := result.(float64)
+	if !ok {
+		return nil, fmt.Errorf("maxFeeCapExpression evaluated to non-float64: %v", result)
+	}
```

Comment:
Since all the inputs to `p.maxFeeCapExpression.Evaluate(parameters)` are typecasted to float64 (actually even one of them being float64 is enough), the result of the function for any mathematical formula is either `float64, nil` or a non-nil error, so ok would always be true. 
https://github.com/OffchainLabs/nitro/pull/1993#discussion_r1415316471

Reply:
I've added a comment that this shouldn't be reachable, but I think it's probably still worth having this check so we would get a good error should it occur.